homeassistant/components/home_connect/sensor.py

```diff
@@ -38,47 +42,206 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+EVENT_OPTIONS_TRANSLATION_KEYS = {
+    BSH_EVENT_PRESENT_STATE_PRESENT: "present",
+    BSH_EVENT_PRESENT_STATE_CONFIRMED: "confirmed",
+    BSH_EVENT_PRESENT_STATE_OFF: "off",
+}
+
+
 @dataclass(frozen=True, kw_only=True)
 class HomeConnectSensorEntityDescription(SensorEntityDescription):
     """Entity Description class for sensors."""
 
-    device_class: SensorDeviceClass | None = SensorDeviceClass.ENUM
-    options: list[str] | None = field(
-        default_factory=lambda: ["confirmed", "off", "present"]
-    )
-    desc: str
-    appliance_types: tuple[str, ...]
+    options_translation_keys: dict[str, str] | None = None
+    default_value: str | None = None
+    appliance_types: tuple[str, ...] | None = None
+    sign: int = 1
+
 
+BSH_PROGRAM_SENSORS = (
+    HomeConnectSensorEntityDescription(
+        key="BSH.Common.Option.RemainingProgramTime",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        sign=1,
+        translation_key="remaining_program_time",
+    ),
+    HomeConnectSensorEntityDescription(
+        key="BSH.Common.Option.Duration",
+        device_class=SensorDeviceClass.DURATION,
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+        sign=1,
+        translation_key="duration",
+    ),
+    HomeConnectSensorEntityDescription(
+        key="BSH.Common.Option.ProgramProgress",
+        native_unit_of_measurement=PERCENTAGE,
+        sign=1,
+        translation_key="program_progress",
+    ),
+)
+
+SENSORS = (
+    HomeConnectSensorEntityDescription(
+        key=BSH_OPERATION_STATE,
+        device_class=SensorDeviceClass.ENUM,
+        translation_key="operation_state",
+        options_translation_keys={
+            "BSH.Common.EnumType.OperationState.Inactive": "inactive",
+            "BSH.Common.EnumType.OperationState.Ready": "ready",
+            "BSH.Common.EnumType.OperationState.DelayedStart": "delayed_start",
+            BSH_OPERATION_STATE_RUN: "run",
+            BSH_OPERATION_STATE_PAUSE: "pause",
+            "BSH.Common.EnumType.OperationState.ActionRequired": "action_required",
+            BSH_OPERATION_STATE_FINISHED: "finished",
+            "BSH.Common.EnumType.OperationState.Error": "error",
+            "BSH.Common.EnumType.OperationState.Aborting": "aborting",
```

Comment:
Okay so all the options I have seen in this PR so far could be just cut off on the last period and then slugified, wdyt?

Reply:
Oh, I didn't know that there was a helper for that at home assistant utils! That can help a lot for sure!

Reply:
Slugify doesn't mean convert to snake-caps
Okey hahahahaha